Contributing.md

```diff
@@ -11,4 +11,33 @@ See https://eclipse.org/legal/ECA.php for details
 
 # Eclipse
 
-The latest Buildship plugin  (>= 2.*) for Eclipse is highly recommended.
+To develop with Eclipse this is the recommended flow:
+
+1. Using the latest [Eclipse](https://www.eclipse.org/downloads/) install a Java development environment that contains these features. These are all available in the "Install New Software" under the name of the release you are using (e.g. Oxygen)
+- JDT The Java Development Tools
+- The latest Buildship plugin  (>= 2.\*) 
+  - The LSP4J build is gradle based
+- Xtend IDE
+  - Much of the code and tests are written in [Xtend](https://www.eclipse.org/xtend/)
+
+2. Fork/Clone the LSP4J repository
+
+3. Import the root of LSP4J as a Gradle project:
+  - File -> Import
+  - Existing Grade Project
+  - Select root of cloned directory as "Project root directory"
+  - Finish the wizard - this will import all the projects but there will be lots of errors
+
+4. Run the "eclipse assemble" gradle targets.
+  - There is already a launch configuration called "lsp4j-build-gradle", run it.
```

Comment:
`There is already a launch configuration called "lsp4j-build-gradle", run it.` -> `There is already a launch configuration in the root of the repository, called "lsp4j-build-gradle", run it.`